utils/metrics.py

```diff
@@ -0,0 +1,192 @@
+import itertools
+
+import numpy as np
+import ot
+import polars as pl
+from scipy.spatial.distance import cdist
+
+
+def _generate_on_off_profiles(
+    profiles: pl.DataFrame, on_signature: list[str], off_signature: list[str]
+):
+    """Generate on and off profiles from the given profiles.
+
+    Parameters
+    ----------
+    profiles : pl.DataFrame
+        The input profiles DataFrame.
+    on_signature : list[str]
+        Morphological profiles that are in the on-morphology signature.
+    off_signature : list[str]
+        The list of features to include in the off profile.
+
+    Returns
+    -------
+    tuple
+        A tuple containing two DataFrames: the on profile and the off profile.
+    """
+    on_profiles = profiles[on_signature]
+    off_profiles = profiles[off_signature]
+    return on_profiles, off_profiles
+
+
+def earths_movers_distance(
+    ref_profiles: pl.DataFrame,
+    exp_profiles: pl.DataFrame,
+    on_signature: list[str],
+    off_signature: list[str],
+    distance_metric: str = "euclidean",
```

Comment:
Consider using a literal type hint here.
```suggestion
    distance_metric: Literal["euclidean"] = "euclidean",
```